polytracker/src/polytracker/output.cpp

```diff
@@ -712,8 +678,12 @@ sqlite3 *db_init(const std::string &db_path) {
   createDBTables(output_db);
   storeVersion(output_db);
   prepSQLInserts(output_db);
+  sqlite3_exec(output_db, "BEGIN TRANSACTION;", NULL, NULL, &errorMessage);
   return output_db;
-  // sqlite3_exec(output_db, "BEGIN TRANSACTION", NULL, NULL, &errorMessage);
 }
 
-void db_fini(sqlite3 *output_db) { sqlite3_close(output_db); }
+void db_fini(sqlite3 *output_db) {
+  char *errorMessage;
+  sqlite3_exec(output_db, "COMMIT;", NULL, NULL, &errorMessage);
```

Comment:
Memory leak if any error. Caller is responsible for freeing memory.

Reply:
Good point! I think before merging ill look over the rest of the code to free any unused allocations. This might only be an issue when errors occur, but its worth handling.